scripts/run-plugin-nuts.js

```diff
@@ -0,0 +1,183 @@
+const got = require('got');
+
+const circleciBaseUrl = 'https://circleci.com/api/v2/';
+
+/**
+ * Script to locate and launch NUTs across sfdx modules
+ *
+ * Script requires that the env var CIRCLECI_API_TOKEN is set to a valid CircleCi API Token
+ *
+ * args:
+ *   sfdx-cli version: z.y.z | latest | latest-rc
+ *
+ * Flow:
+ *   load sfdx-cli package.json
+ *   For each entry in oclif.plugins
+ *     Identify repo organization and project i.e. @salesforce/plugin-auth -> salesforcecli/plugin-auth via "npm show @salesforce/plugin-auth homepage"
+ *     Identify if plugin supports script test:nuts i.e. "npm show @salesforce/plugin-auth scripts | grep test:nuts"
+ *     Check if org/project is known in circleci
+ *       Trigger test nuts workflow for plugin with required parameters
+ */
+
+const manualRepoOverrides = {
+  'salesforce-alm': {
+    org: 'salesforce-cli',
+    repo: 'toolbelt',
+  },
+};
+
+const hasTestNuts = (module) => {
+  const results = execSync(`npm show ${module} scripts`);
+  return results.includes('test:nuts');
+};
+
+const getNpmDependencies = (module) => {
+  const results = execSync(`npm show ${module} dependencies --json`);
+  return JSON.parse(results);
+};
+
+const getOrgAndProject = (module) => {
+  const results = execSync(`npm show ${module} homepage`);
+  if (Reflect.ownKeys(manualRepoOverrides).includes(module)) {
+    return manualRepoOverrides[module];
+  }
+  const repoUrl = new URL(results.toString().trim());
+  const [, org, repo] = repoUrl.pathname.split('/');
+  return { org, repo };
+};
+
+const orgSlug = (org, repo) => {
+  return `github/${org}`;
+};
+const projectSlug = (org, repo) => {
+  return `${orgSlug(org)}/${repo}`;
+};
+
+/**
+ * Triggers the just-nuts workflow for the named org/repo/branch
+ * @param sfdxVersion
+ * @param org
+ * @param repo
+ * @param branch
+ * @returns {Promise<*>}
+ */
+const triggerNutsForProject = async (sfdxVersion, org, repo, branch = 'main') => {
+  const body = {
+    branch: branch,
+    parameters: {
+      'run-auto-workflows': false,
+      'run-just-nuts': true,
+      sfdx_version: sfdxVersion,
+    },
+  };
+  return circle(`${circleciBaseUrl}project/${projectSlug(org, repo)}/pipeline`, { method: 'POST', json: body });
+};
+
+/**
+ * Triggers the nuts monitoring workflow
+ * @param jobData
+ * @param branch
+ * @returns {Promise<*>}
+ */
+const triggerNutsMonitor = async (jobData, branch = 'main') => {
+  const body = {
+    branch: branch,
+    parameters: {
+      'run-auto-workflows': false,
+      'run-just-nuts': true,
+      nuts_job_data: JSON.stringify(jobData),
+    },
+  };
+  return circle(`${circleciBaseUrl}project/${projectSlug('salesforcecli', 'sfdx-cli')}/pipeline`, {
+    method: 'POST',
+    json: body,
+  });
+};
+/**
+ * Makes a request to named URL. Ensures that auth token is applied.
+ * @param url
+ * @param options
+ * @returns {Promise<any>}
+ */
+const circle = async (url, options = {}) => {
```

Comment:
If these scripts grow anymore, we should probably break out similar functionality and require it in.